src/Scan.js

```diff
@@ -116,7 +116,7 @@ class Scan extends React.Component {
 
   state = {};
   store = this.props.stripes.store;
-  checkInRef = React.createRef();
+  barcodeRef = null;
```

Comment:
Could you please explain the reason of changing the way the ref is initialized? I think `React.createRef()` is a better choice when it comes to managing refs.

Reply:
In Scan.js  component CheckIn is a parent component, and it passes callback-ref as prop to child (Field in Checkin.js), and this Field pass into this callback it's barcodeEl, which is created with React.createRef(). So prop is lifting up to parent for having opportunity to set focus to child from parent element, where manipulating with modals are implemented.

Reply:
You can possibly initialize the ref within Scan.js, and pass down as a prop into Checkin, where it can be used like any other prop. Pardon the roughness on this example, I hope it can clarify:

```
<Scan.js>
   const barcodeRef = React.createRef(); 
   focusInput = () => { this.barcodeRef.current.focus() }

   <Checkin.js barcodeRef={barcodeRef}>
         focusInput = () => { this.props.barcodeRef.current.focus()}
        <Field inputRef={this.props.barcodeRef} .../>
    </Checkin.js>
</Scan.js>
```
You can also eliminate the `focusInput` method within Checkin, and just pass the focusInput from Scan as a prop as well. Then simply call the prop in Checkin's `componentDidMount`

This can clean up the code a great deal without the need for getters/setters as well... less code = better coverage by default 🙌

Reply:
I also notice line 246 of Scan.js where it is reaching into a ref to Checkin to call the focus method... That's something that will need cleaning up if you do decide to get rid of the focus handling function (These functions, refs are best to just come from one place!)

Reply:
@JohnC-80, thanks for your comments. I agree that initialize ref in parent component and pass it to child is better and clearer solution